### DISCUSSION TITLE
TPM and AMD SEV support

### DISCUSSION THREAD

**Discussion by ghost** [cloud-hypervisor/cloud-hypervisor#4633] [Q&A] [answered]
As this project looks like a really promising alternative to QEMU for me, I'd like to ask if TPMs (Emulated via swtpm) are supported.

Further I'd like to ask if AMD SEV is supported. I have seen that Intel's TDX is supported, but for SEV haven't found anything.

If one of the both is supported, could you please provide an configuration example or lead me to the docs if any exists?

Thank you for your feedback

**Comment by sboeuf** [comment, marked as answer]
- vTPM is going to be supported at some point. Please follow the status on the issue #2343. @praveen-pk is actively working on this and he has some code on the following branch: https://github.com/praveen-pk/cloud-hypervisor/commits/upstream/vtpm_module/prep
- AMD SEV is currently not supported. But if someone wants to work on this feature, we welcome such contribution :)
